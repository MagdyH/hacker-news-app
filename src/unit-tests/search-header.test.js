import React from 'react';
import renderer from 'react-test-renderer';
import SearchHeader from '../components/search-header';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from "react-redux";

const initialState = {}
const mockStore = configureStore();
const store = mockStore(initialState);


describe('Search Header component', () => {
    it('should render snapshot', () => {
        const component = renderer.create(<Provider store={store}><SearchHeader /></Provider>);

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});