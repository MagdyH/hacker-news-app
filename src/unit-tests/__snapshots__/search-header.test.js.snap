// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Header component should render snapshot 1`] = `
<header
  className="fixed "
>
  <nav
    className="SearchHeader_search"
  >
    <a
      className="Search-Header-Icon"
    >
      <img
        alt=""
        className="header-img"
        src="logo-hn-search-a822432b.webp"
      />
      <div
        className="Search-Header-Label"
      >
        Search 
        <br />
         Hacker News
      </div>
    </a>
    <div
      className="Search-container"
    >
      <span
        className="Search-Icon"
      >
        <i
          aria-hidden="true"
          className="fa fa-search"
        />
      </span>
      <input
        aria-label="Search"
        className="Search-Input"
        onChange={[Function]}
        placeholder="Search stories by title, url or author"
        type="text"
      />
    </div>
  </nav>
</header>
`;
